Sources/LoggingTestKit/TestLogHandler.swift

```diff
@@ -0,0 +1,320 @@
+        return dateFormatter
+    }
+
+    public var timestamp: Date
+    public var label: String
+
+    public var level: Logger.Level
+    public var message: Logger.Message
+    public var metadata: Logger.Metadata
+    public var source: String
+
+    public var file: String
+    public var function: String
+    public var line: UInt
+
+    public var prettyMetadata: String {
+        metadata.sorted(by: {
+            $0.key < $1.key
+        }).map {
+            "\($0)=\($1)"
+        }.joined(separator: " ")
+    }
+
+    public var description: String {
+        "\(Self.timestampFormat.string(from: timestamp)) \(level) \(label) :\(prettyMetadata) [\(source)] \(message)"
+    }
+
+    public init(
+        timestamp: Date = Date(),
+        label: String = "nil",
+        level: Logger.Level = .info,
+        message: Logger.Message = .init(stringLiteral: ""),
+        metadata: Logger.Metadata = [:],
+        source: String = "nil",
+        file: String = "nil",
+        function: String = "nil",
+        line: UInt = 0
+    ) {
+        self.timestamp = timestamp
+        self.label = label
+        self.level = level
+        self.message = message
+        self.metadata = metadata
+        self.source = source
+        self.file = file
+        self.function = function
+        self.line = line
+    }
+
+    // Easily check if log message matches certain characteristics:
+    @available(macOS 13.0, *)
+    public func match(
+        label: String? = nil,
+        level: Logger.Level? = nil,
+        message: Regex<AnyRegexOutput>? = nil,
+        metadata: [(Logger.Metadata.Element) -> Bool] = [],
+        source: Regex<AnyRegexOutput>? = nil,
+        file: Regex<AnyRegexOutput>? = nil,
+        function: String? = nil,
+        line: UInt? = nil
+    ) -> Bool {
+        if let label,
+           self.label != label
+        {
+            return false
+        }
+        if let level,
+           self.level != level
+        {
+            return false
+        }
+        if let message,
+           !self.message.description.contains(message)
+        {
+            return false
+        }
+        for predicate in metadata {
+            if !self.metadata.contains(where: predicate) {
+                return false
+            }
+        }
+        if let source,
+           !self.source.contains(source)
+        {
+            return false
+        }
+        if let file,
+           !self.file.contains(file)
+        {
+            return false
+        }
+        if let function,
+           self.function != function
+        {
+            return false
+        }
+        if let line,
+           self.line != line
+        {
+            return false
+        }
+
+        return true
+    }
+}
+
+public struct TestLogHandler: LogHandler, Sendable {
+    public typealias Filter = (CapturedLogMessage) -> Bool
+
+    public final class Container: @unchecked Sendable {
```

Comment:
I am not yet convinced that we need this type or concept. The thing that I am most worried about is that the container is a reference type. I can understand that after a test you would want to filter all the logged messages by a predicate and then assert on top of the filtered messages but this type is live-updated.

How strongly do you think that this type is needed?